client-device-connection-infinispan/src/main/java/org/eclipse/hono/deviceconnection/infinispan/client/DeviceConnectionInfo.java

```diff
@@ -47,6 +48,28 @@
      */
     Future<Void> setLastKnownGatewayForDevice(String tenant, String deviceId, String gatewayId, Span span);
 
+    /**
+     * For a given list of device and gateway combinations, sets the gateway as the last gateway that acted on behalf
+     * of the device.
+     * <p>
+     * If a device connects directly instead of through a gateway, the device identifier itself is to be used as
+     * gateway value.
+     *
+     * @param tenant The tenant that the device belongs to.
+     * @param deviceIdToGatewayIdMap The map containing device identifiers and associated gateway identifiers. The
+     *                               gateway identifier may be the same as the device identifier if the device is
+     *                               (currently) not connected via a gateway but directly to a protocol adapter.
+     * @param span The active OpenTracing span for this operation. It is not to be closed in this method!
+     *            An implementation should log (error) events on this span and it may set tags and use this span as the
+     *            parent for any spans created in this method.
+     * @return A future indicating the outcome of the operation.
+     *         <p>
+     *         The future will be succeeded if the device connection information has been updated.
+     *         Otherwise the future will be failed with a {@link org.eclipse.hono.client.ServiceInvocationException}.
```

Comment:
IMHO we should also document what happens if only a certain subset of the map's entries could be put to the cache.

Reply:
I don't see details in the Infinispan HotRod client docs on how `remoteCache#putAllAsync` behaves if any of the internal put operations failed.

Reply:
hmm, then maybe we can document that, i.e.
> The outcome is indeterminate if any of the entries cannot be processed by an implementation. In such a case, client code should assume that none of the entries have been updated.